### PR TITLE
Feature/zoom buttons

### PR DESCRIPTION
**Pull request by ayurmedia**
Refactored Code, to be easier to maintain. Moved CSS selectors to top of script. Added many comments to make the script easier to understand. Added opacity 0/1 to prevent flashing of the action-menu while reading/writing to the slider.

### CODE REVIEW

**Comment by ayurmedia** [comment]
The old pull request can be removed and only this one be used. For some reason it shows conflicting files but only this file/version is relevant.

File: Zoom Buttons/ZoomButtons.js [hunk 1]
```diff
@@ -81,121 +92,158 @@ tau
...
+        var closeActionMenuOrKeep = function( was_open ) {
+            if ( !was_open ) {
+                closeActionMenu();
+            }
         };
 
         var openActionMenu = function() {
-            $('.tau-board-header__control--actions .tau-actions-btn:not(.tau-checked)').addClass('zoom-open'); // mark
-            $('.tau-board-header__control--actions .tau-actions-btn:not(.tau-checked)').click(); // open
+            $(zoom_actionbubble_id).css('opacity',0);               // make menu invisible (but open)
```
Comment (tsarevichdm) [inline Zoom Buttons/ZoomButtons.js]: Suggest to save old opacity here and restore on menu close.
So even if we change menu opacity in the future, mashup continue to work.

Comment (ayurmedia) [reply Zoom Buttons/ZoomButtons.js]: OK, I can store the old value in temp and restore it

Comment (ayurmedia) [reply Zoom Buttons/ZoomButtons.js]: The old opacity value is now stored, and restored later

**Comment by ayurmedia** [comment]
Updated baseinfo to have current TP cloud version.

**Comment by tsarevichdm** [comment]
Can you, please, rebase this branch over upstream/master and resolve conflicts?

**Comment by ayurmedia** [comment]
I merged in the changes from origin:master. (https://github.com/TargetProcess/TP3MashupLibrary)  Hopefully this is now a clean pull request. 

Changes from @mdedeiko are also pulled in.

**Pull request merged**